### ISSUE TITLE
Indentation#lineWrappingIndentation is strictly enforced on 'annotation array initialization'

### ISSUE THREAD

**Issue by renierG** [checkstyle/checkstyle#12150]
From the documentation in [Indentation|https://checkstyle.sourceforge.io/config_misc.html#Indentation] on _forceStrictCondition_:

_If value is false, line wrap indent could be bigger on any value user would like._

I would expect, that with a _lineWrappingIndentation_ set to 4 and _forceStrictCondition_ not set (default: _false_) the provided java file should be.

Instead the audit failes on a greater indentation level (here 8).

---

[|https://github.com/renierG/checkstyle-bugreport]

```java
package ladem.shop.middleware.service.shopware.client.schema;

import com.fasterxml.jackson.annotation.JsonSubTypes;

@JsonSubTypes(value = {
        @JsonSubTypes.Type(value = A.class, name = "a"),
        @JsonSubTypes.Type(value = B.class, name = "b"),
})
public abstract class Test {
}
```

```bash
java -jar checkstyle-10.3.3-all.jar -c checkstyle_lineWrappingIndentation4.xml Test.java
Starting audit...
[ERROR] /Users/mgrein/Downloads/checkstyle/Test.java:6:9: 'annotation array initialization' child has incorrect indentation level 8, expected level should be 4. [Indentation]
[ERROR] /Users/mgrein/Downloads/checkstyle/Test.java:7:9: 'annotation array initialization' child has incorrect indentation level 8, expected level should be 4. [Indentation]
Audit done.
Checkstyle ends with 2 errors.
```

```bash
java -jar checkstyle-10.3.3-all.jar -c checkstyle_lineWrappingIndentation8.xml Test.java
Starting audit...
Audit done.
```

**Comment by nrmancuso** [comment]
@renierG please follow issue template and provide your configuration.

**Comment by renierG** [comment]
Check documentation: [Indentation](https://checkstyle.sourceforge.io/config_misc.html#Indentation)

---

```bash
javac Test.java

cat Test.java
package ladem.shop.middleware.service.shopware.client.schema;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Test.JsonSubTypes(value = {
        @Test.JsonSubTypes.Type(),
        @Test.JsonSubTypes.Type()
})
public class Test {

    @Target({ElementType.ANNOTATION_TYPE, ElementType.TYPE, ElementType.FIELD, ElementType.METHOD, ElementType.PARAMETER})
    @Retention(RetentionPolicy.RUNTIME)
    public @interface JsonSubTypes {
        JsonSubTypes.Type[] value();

        @interface Type {
        }
    }
}
```

```bash
cat checkstyle_lineWrappingIndentation4.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name = "Checker">
    <module name="TreeWalker">
        <module name="Indentation">
            <property name="basicOffset" value="4"/>
            <property name="braceAdjustment" value="0"/>
            <property name="caseIndent" value="4"/>
            <property name="throwsIndent" value="8"/>
            <property name="lineWrappingIndentation" value="4"/>
            <property name="arrayInitIndent" value="4"/>
        </module>
    </module>
</module>

java -jar checkstyle-10.3.3-all.jar -c checkstyle_lineWrappingIndentation4.xml Test.java
Starting audit...
[ERROR] /Users/mgrein/Downloads/checkstyle/Test.java:9:9: 'annotation array initialization' child has incorrect indentation level 8, expected level should be 4. [Indentation]
[ERROR] /Users/mgrein/Downloads/checkstyle/Test.java:10:9: 'annotation array initialization' child has incorrect indentation level 8, expected level should be 4. [Indentation]
Audit done.
Checkstyle ends with 2 errors.
```

```bash
cat checkstyle_lineWrappingIndentation8.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name = "Checker">
    <module name="TreeWalker">
        <module name="Indentation">
            <property name="basicOffset" value="4"/>
            <property name="braceAdjustment" value="0"/>
            <property name="caseIndent" value="4"/>
            <property name="throwsIndent" value="8"/>
            <property name="lineWrappingIndentation" value="8"/>
            <property name="arrayInitIndent" value="4"/>
        </module>
    </module>
</module>

java -jar checkstyle-10.3.3-all.jar -c checkstyle_lineWrappingIndentation8.xml Test.java
Starting audit...
Audit done.
```

---

Expected: with _lineWrappingIndentation_ set to 4 and _forceStrictCondition_ not set (default: _false_) the provided java file should be valid for both config files.
